logging_middleware.go

```diff
@@ -3,21 +3,54 @@ package handlers
 import (
 	"context"
 	"net/http"
+	"regexp"
 	"time"
 
+	"gopkg.in/errgo.v1"
+
 	"github.com/codegangsta/negroni"
 	"github.com/sirupsen/logrus"
 )
 
+var (
+	loggerFuncMap = map[logrus.Level]func(logrus.FieldLogger, string, ...interface{}){
+		logrus.DebugLevel: logrus.FieldLogger.Debugf,
+		logrus.InfoLevel:  logrus.FieldLogger.Infof,
+		logrus.WarnLevel:  logrus.FieldLogger.Warnf,
+		logrus.ErrorLevel: logrus.FieldLogger.Errorf,
+		logrus.FatalLevel: logrus.FieldLogger.Fatalf,
+		logrus.PanicLevel: logrus.FieldLogger.Panicf,
+	}
+)
+
+type patternInfo struct {
+	re    *regexp.Regexp
+	level logrus.Level
+}
+
 type LoggingMiddleware struct {
-	logger logrus.FieldLogger
+	logger  logrus.FieldLogger
+	filters map[string]patternInfo
 }
 
 func NewLoggingMiddleware(logger logrus.FieldLogger) Middleware {
-	m := &LoggingMiddleware{logger}
+	m := &LoggingMiddleware{logger: logger, filters: map[string]patternInfo{}}
 	return m
 }
 
+func NewLoggingMiddlewareWithFilters(logger logrus.FieldLogger, filters map[string]logrus.Level) (*LoggingMiddleware, error) {
+	refilters := map[string]patternInfo{}
+	for pattern, level := range filters {
+		re, err := regexp.Compile(pattern)
+		if err != nil {
+			return nil, errgo.Notef(err, "invalid regexp '%v'", pattern)
+		}
+		refilters[pattern] = patternInfo{re: re, level: level}
```

Comment:
No possible to use the regexp as key, as a pointer can't be a key in a map. (And a regexp should be a pointer as there is a mutex in it)